v2/authorisation/middleware.go

```diff
@@ -175,3 +189,17 @@ func (m PermissionCheckMiddleware) Close(ctx context.Context) error {
 func (m PermissionCheckMiddleware) HealthCheck(ctx context.Context, state *health.CheckState) error {
 	return m.permissionsChecker.HealthCheck(ctx, state)
 }
+
+// GetCollectionIdAttribute provides an implementation of GetAttributesFromRequest. Retrieves and returns
+// header 'Collection-Id' from the request if it exists, otherwise returns an empty map. Never returns an
+// error as the header is not mandatory
+func GetCollectionIdAttribute(req *http.Request) (attributes map[string]string, err error) {
+	attributes = map[string]string{}
```

Comment:
You don't need to declare this here as the func signature specifies a _named return parameter_ You can use `attributes` without initialising it.

Reply:
I want an empty map as opposed to a nil map here hence the initialisation

Reply:
In that case could we refactor this to remove the named returns - I don't think it adds any value in this case, the same can be achieved with:
```
func GetCollectionIdAttribute(req *http.Request) (map[string]string, error) {
	attributes := make(map[string]string, 0)

	collectionIdAttribute, _ := headers.GetCollectionID(req)
	if collectionIdAttribute != "" {
		attributes[collectionIdAttributeKey] = collectionIdAttribute
	}

	return attributes, nil

```

Reply:
done